### PR TITLE
Extend user profile editor to allow editing of bio, address, etc.

### LINKED ISSUE #26: User Profile Edit Page - FE
As a registered user, I can navigate to my profile page `/user/profile`, and click the edit icon.
Then I am on my profile edit page `/user/profile/edit`.

I see pre-populated fields for:
- Email address

- [x] Add profile edit page `/user/profile/edit`

### PR DESCRIPTION
**Pull request by rubyr**
## Type of change made:
- [ ] **Bugfix** (non-breaking change which fixes an issue)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [ ] **Refactoring** (improves existing functionality)
- [x] **Testing** (tests existing code or improves existing tests)
- [ ] **Styling** (no functionality changed, just visual improvements)

## Description:

This PR adds a ProfileEditor component, to allow the user to change their profile information. It also updates the Profile component to display extra information on the page, such as their email, address, and biography, as well as an image. 

## What functionality does this accomplish?

- Allows user to edit all of their profile info
- Displays user info on profile page
- Updates existing user queries to include extra info from DB

closes #26

## Checklist:
- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have fully tested my code
- [x] I have partially tested my code (please explain why):
    - Need profileeditor tests

### CODE REVIEW

File: code/web/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        "tdd",
+        "--timeout",
+        "999999",
+        "--colors",
+        "${workspaceFolder}/test"
+      ],
+      "internalConsoleOptions": "openOnSessionStart",
+      "name": "Mocha Tests",
+      "program": "${workspaceFolder}/node_modules/mocha/bin/_mocha",
+      "request": "launch",
```
Comment (J-Poulter) [inline code/web/.vscode/launch.json:16]: What are these lines doing?  Is it setting up the testing environment

**Comment by SMJ289** [comment]
Does this PR also add the ability to display the newly added information on the User Profile page (bio, address, image)? I don't see that in the PR description.

**Comment by rubyr** [comment]
It does! Let me add that.

**Comment by SMJ289** [comment]
I think it would be a good idea to be a little more detailed in general in regard to the description and functionality sections of the PR template.

**Comment by SMJ289** [comment]
If we are adding testing should we also indicate that using the provided checkbox?

File: code/web/src/modules/user/ProfileEditor.test.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import jest from "jest";
+import ProfileEditor from "./ProfileEditor"
+import { render } from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect"
+import { Provider } from "react-redux";
+import { createStore } from "redux";
+
+describe('ProfileEditor', () => {
```
Comment (J-Poulter) [inline code/web/src/modules/user/ProfileEditor.test.js:9]: I think you're on the right track for setting up these tests in Redux.  I'll share my redux testing code I did on another project to show you how I went about testing this stuff.

File: code/web/src/modules/user/api/actions.test.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import * as userAction from "./actions";
+
+describe('User Actions', () => {
```
Comment (J-Poulter) [inline code/web/src/modules/user/api/actions.test.js:3]: Thanks for getting us started on testing these actions.  This is pretty much the way they taught us to test Redux actions back when they were still testing it.

File: code/web/.vscode/launch.json (hunk 1 above)
Comment (rubyr) [reply code/web/.vscode/launch.json:16]: I forgot to remove this file, I thought I had captured it in the .gitignore. It was supposed to set up testing within the VSCode debugger, but it did not work, unfortunately.

File: code/web/package.json [hunk 4]
```diff
@@ -12,7 +12,8 @@
     "build:client:prod": "webpack -p",
     "build:server:prod": "babel src -s -D -d build",
     "start:server:prod": " node build/index.js",
-    "storybook": "start-storybook -p 9000 -c storybook"
+    "storybook": "start-storybook -p 9000 -c storybook",
+    "test": "jest --watch"
```
Comment (kmcgrevey) [inline code/web/package.json:16]: using jest in the `api` directory as well

**Pull request merged**